samples/System.Numerics.Tensors.Samples/TensorSamples/ReversePInvoke.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Numerics;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace TensorSamples
+{
+    partial class ReversePInvoke
+    {
+        public static void RunSample()
+        {
+            Console.WriteLine($"*** {nameof(ReversePInvoke)} ***");
+            var multTable = GetMultiplicationTable(5);
+
+            Console.WriteLine("Multiplication table:");
+            Console.WriteLine(multTable.GetArrayString());
+
+            Console.WriteLine("Sums:");
+            for (int row = 0; row < multTable.Dimensions[0]; row++)
+            {
+                Console.WriteLine(GetRowSum(multTable, row));
+            }
+        }
+        
+        // The following represent what .NET bindings would look like for a native library that wishes
+        // to deal in tensors and allocate them on the managed heap through callbacks.
+        
+        public static DenseTensor<double> GetMultiplicationTable(int maxNumber)
+        {
+            using (var allocator = new Allocator<double>())
+            {
+                var bufferPtr = GetMultTableAllocateManaged(allocator.AllocateArray, maxNumber);
+
+                var buffer = allocator.Unpin(bufferPtr);
+
+                var result = new DenseTensor<double>(buffer, new[] { maxNumber, maxNumber });
+
+                return result;
+            }
+        }
+
+        public static unsafe double GetRowSum(DenseTensor<double> tensor, int row)
+        {
+            fixed(double* dataPtr = &tensor.Buffer.Span.DangerousGetPinnableReference())
+            {
+                return GetRowSum(dataPtr, tensor.Dimensions.ToArray(), tensor.Rank, row);
```

Comment:
Same here. Why `tensor.Dimensions.ToArray()` and not fix the `Dimensions` span? No need to `alloc. GetRowSum` should be changed accordingly.